### DISCUSSION TITLE
Problems getting own mesh running

### DISCUSSION THREAD

**Discussion by Otherside5** [mdolab/dafoam#271] [General]
Hello everyone,

I am trying to optimize a pump turbine with a rough 170k cell mesh based on the Rotor37 runScript with DASimpleFoam.

The simulation is running without an issue with the normal simpleFoam solver (in the same docker image) and checkMesh does not raise any issues, but when I try to run the simulation with the task "runPrimal" of the runScript file I get something like the message below. I tried different versions of DAFoam, different fvSchemes and fvSolution settings but it does not change anything.

[logOpt.txt](https://github.com/mdolab/dafoam/files/8112310/logOpt.txt)

Do you have any guess what I am doing wrong?

Thank you
Alex

```
Starting time loop

Checking mesh quality for time = 0
    Overall domain bounding box (nan -nan nan) (nan -nan nan)
    Mesh has 3 geometric (non-empty/wedge) directions 3{1}
    Mesh has 3 solution (non-empty) directions 3{1}
    Boundary openness (-nan nan nan) OK.
    Max cell openness = nan OK.
    Max aspect ratio = nan OK.
    Minimum face area = nan. Maximum face area = nan.  Face area magnitudes OK.
    Min volume = nan. Max volume = nan.  Total volume = nan.  Cell volumes OK.
    Mesh non-orthogonality Max: -nan average: -nan
    Non-orthogonality check OK.
    Face pyramids OK.
    Max skewness = nan OK.
    Coupled point location match (average nan) OK.

Mesh OK.

Time = 1

DILUPBiCG:  Solving for Ux, Initial residual = nan, Final residual = nan, No Iterations 8
DILUPBiCG:  Solving for Uy, Initial residual = nan, Final residual = nan, No Iterations 8
DILUPBiCG:  Solving for Uz, Initial residual = nan, Final residual = nan, No Iterations 8
U Initial residual: (nan nan nan)
U   Final residual: (nan nan nan)
DICPCG:  Solving for p, Initial residual = nan, Final residual = nan, No Iterations 1000
p Initial residual: nan
p   Final residual: nan
DICPCG:  Solving for p, Initial residual = nan, Final residual = nan, No Iterations 1000
[2] 
[2] 
[2] --> FOAM FATAL IO ERROR: 
[2] Wrong token type - expected scalar value, found on line 0: word 'nan'
[2] 
[2] file: /home/dafoamuser/mount/processor2/system/data.solverPerformance.p at line 0.
[2] 
[2]     From function Foam::Istream& Foam::operator>>(Foam::Istream&, Foam::doubleScalar&)
[2]     in file lnInclude/Scalar.C at line 149.
[2] 
FOAM parallel run exiting
```

**Comment by friedenhe** [comment]
Your FFD did not fully cover the design surfaces.

**Reply by Otherside5** [reply]
Thank you that was the problem.

**Comment by Otherside5** [comment]
Hey,

the optimization is running know but I get the error:

```
Solving Linear Equation... 573 s
Main iteration 0 KSP Residual norm 2.704290667506e-03 603 s
Main iteration 0 KSP Residual norm 2.704290667506e-03 603 s 
Total iterations 0
Solving Linear Equation... Completed! 603 s
Residual tolerance not satisfied, solution failed!
```

It carries on with the optimization after that but the never tries to solve the linear equation system.

Do you have any idea what could be the issue?

Thanks
Alex

**Reply by friedenhe** [reply]
Which objective function and boundary conditions did you use?

**Reply by Otherside5** [reply]
objective is to maximize moment

BC:
[BC.txt](https://github.com/mdolab/dafoam/files/8147397/BC.txt)

nothing special except maybe cylindricalInletVelocity but I have used this before with DAFoam, and fixedMean but replacing this BC does not change anything.

Thanks for your help.

**Comment by friedenhe** [comment]
Hmm, this usually happens when you use a new objective function or boundary conditions. I suggest you start with a working case and then gradually change to your current setup and see which change breaks the case.
